backend/src/main/kotlin/com/infowings/catalog/data/objekt/ObjectValidator.kt

```diff
@@ -236,10 +249,40 @@ class MainObjectValidator(
             objectPropertyVertex,
             aspectPropertyVertex,
             parentValueVertex,
-            valueVertex.measure
+            measureVertex
         )
     }
 
+    private fun recalculateValueAccordingToMeasure(originalValue: ObjectValueData, measure: Measure<DecimalNumber>?) =
+        if (originalValue is ObjectValueData.DecimalValue && measure != null) {
+            val valueRepresentation = BigDecimal(originalValue.valueRepr)
+            ObjectValueData.DecimalValue(measure.toBase(DecimalNumber(valueRepresentation)).value.toString())
+        } else {
+            originalValue
+        }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun validateMeasureInRequest(measureGroup: MeasureGroup<*>?, requestMeasureName: String?): Measure<DecimalNumber>? {
+        val measure = when {
+            measureGroup != null && requestMeasureName != null ->
+                measureGroup.measureList.find { it.name == requestMeasureName }
+                        ?: throw IllegalArgumentException("Measure group for value (${measureGroup.name}) does not have measure $requestMeasureName")
```

Comment:
Let's keep MeasureGroup details encapsulated and define method MeasureGroup.find(measureName: String) instead of dealing with implementaton details outside.